src/app/clusters/bindings/BindingManager.h

```diff
@@ -123,22 +121,46 @@ class BindingManager
     static BindingManager & GetInstance() { return sBindingManager; }
 
 private:
-    static BindingManager sBindingManager;
-
-    static void HandleDeviceConnected(void * context, Messaging::ExchangeManager & exchangeMgr, SessionHandle & sessionHandle);
-    void HandleDeviceConnected(Messaging::ExchangeManager & exchangeMgr, SessionHandle & sessionHandle);
+    class ConnectionCallback
+    {
+    public:
+        ConnectionCallback(BindingManager * bindingManager) :
+            mBindingManager(bindingManager), mOnConnectedCallback(HandleDeviceConnected, this),
+            mOnConnectionFailureCallback(HandleDeviceConnectionFailure, this)
+        {}
+
+        Callback::Callback<OnDeviceConnected> * GetOnDeviceConnected() { return &mOnConnectedCallback; }
+        Callback::Callback<OnDeviceConnectionFailure> * GetOnDeviceConnectionFailure() { return &mOnConnectionFailureCallback; }
+
+    private:
+        static void HandleDeviceConnected(void * context, Messaging::ExchangeManager & exchangeMgr, SessionHandle & sessionHandle)
+        {
+            ConnectionCallback * connectionCallback = static_cast<ConnectionCallback *>(context);
```

Comment:
Suggestion: I tend to follow a practice of naming this `_this` so that it better conveys that we're essentially getting back a pointer to 'ourself'. That then makes the delete later clear as well, that we're actually deleting ourselves from within a static method in the same class...